INSVCTwoFluidStaggeredHierarchyIntegrator.cpp

```diff
@@ -494,10 +497,36 @@ INSVCTwoFluidStaggeredHierarchyIntegrator::integrateHierarchy(const double curre
         } // patches
     }     // levels
 
+    switch(d_viscous_time_stepping_type) {
+        case TRAPEZOIDAL_RULE:
+            VCTwoFluidStaggeredStokesOperator stokes_op("stokes_op", true);
+            const double D1 = 0.5; // This will depend on the time stepping scheme.
+            const double D2 = -0.5; // This will depend on the time stepping scheme. 
+
+            stokes_op.setCandDCoefficients(0.0, D1);
+            stokes_op.setDragCoefficient(1.0, 1.0, 1.0);
+            stokes_op.setViscosityCoefficient(1.0, 1.0);
+            stokes_op.setThnIdx(thn_cc_idx);
+            
+            // store results of applying stokes operator in f2_vec
+            Pointer<SAMRAIVectorReal<NDIM, double>> f2_vec;
+            f2_vec = f_vec.cloneVector("f2_vec");
+            f2_vec->allocateVectorData();
+            f2_vec->copyVector(Pointer<SAMRAIVectorReal<NDIM, double>>(&f_vec, false), true);
+            stokes_op.initializeOperatorState(u_vec, *f2_vec);
+            stokes_op.apply(u_vec, *f2_vec);
+
+            // Sum f_vec and f2_vec and store result in f_vec
+            Pointer<SAMRAIVectorReal<NDIM, double>> f_vec_ptr(&f_vec, false);
+            f_vec.axpy(1.0, f_vec_ptr, f2_vec, true);
+
+        case BACKWARD_EULER:
+            const double D2 = -1.0; // This will depend on the time stepping scheme.
```

Comment:
Suggest creating two operators, a "solve" operator and a "rhs" operator. Only set the `D1` and `D2` (and the `C` coefficients if necessary) in the switch statement. Then apply the operator outside the switch statement. This way you can remove the for loop above. You also won't have branching code for each time stepping type.

Reply:
I pushed some new changes based on your review. You mentioned creating a "solve" operator, but I wasn't sure what you wanted for that.